Python/specialize.c

```diff
@@ -667,6 +669,17 @@ _Py_Specialize_LoadAttr(PyObject *owner, _Py_CODEUNIT *instr, PyObject *name)
         }
         goto success;
     }
+    int oparg = _Py_OPARG(*instr);
+    if (oparg & 1) {
+        int res = specialize_attr_loadmethod(owner, instr, name);
```

Comment:
Why specialize this differently?

Reply:
I'm unsure sure which part of the code you're referring to. Do you mean why this code only specializes when `oparg & `?

If so, that's because the semantics of `LOAD_METHOD` are slightly different than `LOAD_ATTR` due to unbound/bound method pushed. This behavior is invalid if there isn't a call. So I need to guard it for calls only. The only exception to this I suspect is the current `LOAD_METHOD_CLASS`.

What I mean to say is that, `LOAD_METHOD` is a specific subset of `LOAD_ATTR`. All `LOAD_ATTR` specializations work for `LOAD_METHOD`, but not vice versa.

Reply:
Yes, `LOAD_METHOD` needs to be handled differently in some cases.
But we should determine the case first, then split the behavior depending on `oparg & 1`.

For example, if an attribute is a non-method on a class, then we can treat it the same for both `LOAD_ATTR` and `LOAD_METHOD`. Splitting the specialization into `specialize_attr_loadmethod` and `_Py_Specialize_LoadAttr` means duplicating code and maybe missing some optimization opportunities.

Reply:
Ahh ok I see what you want me to do now.